### ISSUE TITLE
Addon is not cleaning localstorage

### ISSUE THREAD

**Issue by monterrr** [Lusito/forget-me-not#193]
The addon is not cleaning localstorage upon closing tabs.

**Comment by Speravir** [comment]
I must confirm this. In addon settings I have the default _clean on exit_ (or what the name in English is). But In the Firefox preferences I can see, that for quite a lot of pages the local storaged data has not been deleted, despite I have not whitelisted them. Unfortunately I was too fast with manual deleting there, but if I remember correctly the oldest data have been last used about two months ago, so this would would concur with the latest addon changes. I am not sure about this, though!

BTW see also in [Vergiss Mein Nicht (cookies, local storage, etc. löschen) – Seite 3](https://www.camp-firefox.de/forum/thema/123889-vergiss-mein-nicht-cookies-local-storage-etc-l%C3%B6schen/?postID=1128279#post1128279) pp. (at least until my reaction), though a bit confusing in linked posting.

**Comment by Lusito** [comment]
Please check out this explanation:
https://github.com/Lusito/forget-me-not/issues/175#issuecomment-526918057

The essence of it: What firefox tells you is "Storage" is not just local-storage. It's indexed-db, local-storage, meta-data and maybe more. What remains after closing the tabs is meta-data, which can not be accessed by the website anymore.
You can try it by checking http://lusito.info/storage/

**Comment by Speravir** [comment]
Thank you for information. I should have known this.

> You can try it by checking http://lusito.info/storage/

For me it says 
```
Cookie:I don't remember you
Local Storage:I don't remember you
IndexedDB:I don't remember you
```
So it says it works as intended by me.

**Comment by monterrr** [comment]
Something is not right, I get on first page of youtube, videos I watched before despite having closed all tabs and with 1 second delay delete settings, and these videos are not popular and are old to be on the first page and I'm not logged in nor do I have a google account nor do I have other pages open related to youtube or google. This clearly shows that the data firefox reports to still have that doesn't get cleaned is a problem.

**Comment by monterrr** [comment]
Can anyone test youtube?

**Comment by monterrr** [comment]
I have confirmed this, if I go to the firefox preferences and to cookies and site data and remove everything from youtube it stops displaying the videos!

So this is a bug!

**Comment by Lusito** [comment]
I just checked and can't reproduce.

Youtubes landing page shows me only video recommendations based on my previous videos as long as my cookies have not been cleaned. Once cleaning happened, everything gets reset back.

You can try this by using the manual cleanup and only select cookies.
What might happen here is:
If you have youtube open in one tab and another tab that has a youtube video embedded and you close youtube, then the embedded youtube video is still active, so it counts as youtube still being open.

Try these things to see if cleanup of cookies actually happened:
- Enable the setting "Show a notification when cookies have been removed"
  - Then leave youtube and check if the notification lists youtube cookies.
- To double  check, use the cookie inspector (the small book with a circle above it) to check if there are still cookies for youtube.
  - If there are still cookies left, check the badge beside it. if it's not red or black, then these cookies have no rules that make them get cleaned up.

**Comment by monterrr** [comment]
The problem seems to be that I have a tab with a video of youtube, but that tab isn't loaded because I closed Firefox and opened it and it doesn't load the tab until clicking, is this a bug?

**Comment by Lusito** [comment]
Hard to say.. I have never tried a scenario like this. I would have thought, that tabs, which are not loaded, don't list their iframes. Will have to take a deeper look into this. Thanks for the extra information.

Sidenote: #83 and #54 cover the general questions of unloaded tabs.

**Comment by rkuralev** [comment]
I can confirm that the extension has issues with youtube. Neither cookies nor local storage are not cleaned on domain leave.

**Comment by eadamy** [comment]
I have a small whitelist and settings to delete cookies and local storage not whitelisted 10 seconds after leaving domain (attachment 1) The cookies are being cleaned properly. Local storage is not even after exiting FF and restarting. See highlighted items in attachment 2.
![20200115 1030-](https://user-images.githubusercontent.com/25034193/72457001-f7269b00-3782-11ea-8a69-c24aa8238530.png)
![20200115 1033-Options](https://user-images.githubusercontent.com/25034193/72457036-fc83e580-3782-11ea-8f72-52baeb7a86a1.png)

.

**Comment by Lusito** [comment]
beware of what firefox calls "storage" in the firefox options. This is not just local storage, but other store as well (like indexed db, service workers and meta data).

Try this demo to see if your local storage gets deleted: http://lusito.info/storage/

**Comment by eadamy** [comment]
Thank you for the test which appears to show everything deleted for
lusito.info. And thank you for the addon which does a great job with
cookies.

I think the point that those of us writing about this are trying to make is
that we don't care what FF calls it. Unless we flag it "Never", we want
anything deposited by the website (including indexed db, service workers,
and meta data) removed according to the timing of the rule. Just like the
cookies.

**Comment by Lusito** [comment]
Yeah, I hear you. Sadly, there is little I can do at this moment.

**Comment by eadamy** [comment]
Hopefully, an issue you can address with version 3. Certainly understand not undertaking major improvements to the current version. Thanks again.
